### DISCUSSION TITLE
Save Unique Index for Collection by Code Cannot effect on database

### DISCUSSION THREAD

**Discussion by toby1991** [pocketbase/pocketbase#4180] [Q&A] [answered]
```golang
collectionTemplate, err := app.Dao().FindCollectionByNameOrId(tableName)
	if err != nil {
		app.Logger().Error(err.Error(), err)
		return err
	}
	if err := json.Unmarshal([]byte(`[
			"CREATE UNIQUE INDEX `+"`"+`idx_xxxxxx`+"`"+` ON `+"`"+tableName+"`"+` (\n  `+"`"+`goods_id`+"`"+`,\n  `+"`"+`goods_name`+"`"+`,\n  `+"`"+`user_id`+"`"+`\n)"
		]`), &collectionTemplate.Indexes); err != nil {
		return err
	}

	if err := app.Dao().SaveCollection(collectionTemplate); err != nil {
		app.Logger().Error(err.Error(), err)
		return err
	}

	return nil

```

THIS CODE WILL DELETE ALL THE INDEX, AND ADD A UNIQUE INDEX.
THEN THE UNIQUE INDEX WOULD BE SEEN AT POCKETBASE DASHBOARD, AND `_collections` TABLE.
BUT! IT CANNOT BE SEEN AT TABLE SCHEMA IN SQLITE, ALSO, IT CANNOT EFFECTED BY UNIQUE LOGIC!! 

AND THIS CODE WOULD BE SUCCESSFUL WHEN CALLED ONCE IN THE PROGRAM, BUT IF IT IS IN THE `for` LOOP, IT WILL SHOW THE SCENE AS ABOVE

**Comment by ganigeorgiev** [comment]
I don't understand the above description, nor what you are asking for.

Please try to elaborate and/or more provide a more clear example.

**Reply by toby1991** [reply]
Use the code to create unique index for multiple table in a for loop, It cannot create the workable unique index as expected, The unique index can be seen in the pocket base web dashboard and _collection table row for the table, but cannot be seen in the SQLite table schema.
Btw, manual creation of the unique index in the web dashboard works well .

**Reply by toby1991** [reply, marked as answer]
Sorry that's my fault, unique key name in Sqlite is unique in the whole database, not the same as MySql that is in its' table
